README.md

```diff
@@ -31,24 +31,26 @@ When a major issue in "beta 3" is found on SOAK, we will follow the steps above
 Only pre-releases that are SOAKed for at least two weeks should be used as stable releases. After the soaking period was successful, follow these steps:
 
 1. fetch latest repo state: `git fetch`
-1. checkout the pre-release tag you want to make a stable release: e.g. `git checkout v1.1.0-rc.3`
-1. create new `stable-*` branch for that minor release: e.g. `git checkout -b stable-1.1.x`
-1. update kommanders `appVersion` to be stable by removing the pre-release suffix and commit that change to your newly created `stable-*` branch (remember to also update revision)
+1. update kommanders `appVersion` to be stable by removing the pre-release suffix and commit that change to `master` (remember to also update revision)
```

Comment:
I read over the removed lines and thought about this some more.. I guess the previous version is probably correct - the tag should be off the latest rc tag, not master in case there were changes to master after the rc (that shouldn't go into stable tag).

The step to add is to merge the stable branch back into master after tagging - noting that there will be merge conflicts to resolve due to adding a new revision (let's say revision 50 for example) in the stable branch that also exists on master. While resolving conflicts, we should use stable's version of revision 50, and move master's revision 50 and any others after it, into a new dir that is stable version + 0.0.1 (next patch version). I'll try to suggest this step below

I just checked KBA to compare, and they are grouping patches in the same dir (`1.0.x` dirs instead of `1.0.0` for example). We can consider moving to this as well to be in line with them

Reply:
tl;dr
removing these lines and only creating the stable branch after everything is done was correct IMHO.

long
generally speaking, everything "new" should happen on master, the only reason we have `stable-*` _maintenance_ branches is that we sometimes need to patch older versions (e.g. releasing `1.1.1` while master already has `1.2.0-beta.x`) - only in these scenarios we actually _need_ to create a `stable-*` branch for the older version (`stable-1.1.x` in this example). Us doing it earlier is to make it a bit easier to work with the repo.

that said, working on unreleased minor versions ((re- and GA releases), we should mainly/only work on master branch and merging any `stable-*` branch back to master must never result in a conflict. if we get a conflict, something went wrong before and we should revert/fix the root cause instead of trying to duct-tape something with a resolved conflict.


I'd also vote for individual folders per patch version as long as we have the revision concept. we want to be able to version (revision) patch versions indepenendtly.

Reply:
Is it possible that the pre-release we want to release is not the version of the latest master? I think it makes sense that the latest revision in a version (directory) should always be the one that is released so we shouldn't see conflicts in master vs stable, as you mentioned. But would we ever be in a scenario where, prior to releasing a version e.g. 1.1.0, we'd already created a 1.2.0 dir and started developing for a newer version? If we do think that's a possible scenario, then if we were to branch off master/tag the latest master, the 1.2.0 revisions would be present in a 1.1.0 stable branch.

I think it makes sense to work off the rc tags since they are, aptly named, the release candidates that should eventually become the stable release. And would avoid issues if a newer version underwent development prior to the version getting released.

Reply:
what you are describing possibly can happen if we forgot to branch off, in such case we'd probably need to revert all newer 1.1.0 revisions, branch off the latest tag, re-add the previously reverted revisions, and merge back 1.1.0 maintenance branch in master. 

so essentially reverting a mistake, and doing it properly.

Reply:
I still feel like we may run into the situation where we want to start developing the next version before we are ready to release the current version, which might put us in a bad state. But I'm ok just getting this PR in with this version now since this has been sitting around long enough, and if we happen to run into said scenario in the future, we can revisit the process

Reply:
@shaneutt @samvantran any final opinions on the matter? I think otherwise we can get this in!

Reply:
i think this is good for now. like you said, we can revisit if we run into the above scenario

Reply:
> I still feel like we may run into the situation where we want to start developing the next version before we are ready to release the current version, which might put us in a bad state. 

thats possible, we just need to branch off of master before we create the new versions folder. from that point on we can work on multiple branches (just like we do currently for 1.1. & 1.2). only important bit is that we branch off _before_ creating new version folders. that the only important thing to keep in mind.